<a href="https://colab.research.google.com/github/gaemeebum/683380587-7_sec1_script_programming/blob/main/Sec1_Week1_Lab1_683380587_7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Week 1 683380857-7 นางสาวศศิวรรณ ขื่นนิรันดร์



Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello world")

Hello world


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [ ]:
Age = 22
GPA = 3.25
Favorite_Color = "Blue"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  22 <class 'int'>
GPA =  3.25 <class 'float'>
Favorite_Color =  Blue <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [ ]:
num1 = 6
num2 = 7
sum = num1 + num2
print("Sum = ",sum)
sub = num1 - num2
print("Sub = ",sub)
multiply = num1 * num2
print("Mul = ",multiply)
division = num1 / num2
print("Div = ",division)

Sum =  13
Sub =  -1
Mul =  42
Div =  0.8571428571428571


4. String concatenation and f-strings:


In [ ]:
First_name = "Sasiwan"
Last_name = "Chuenniran"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Sasiwan Chuenniran


5. User input and type conversion:


In [ ]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 62
type of A is :  <class 'str'>
62  + 5 =  67


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: ในสถานการณ์จริงเราสามารถนำเอาไปคำนวณหาค่าคำตอบของสมการคณิตศาสตร์ได้ เช่น ต้องการอยากรู้ว่า 67 * 76 ได้เท่าไหร่ เราสามารถเขียนโค้ดกำหนดตัวเลขคำนวณหาคำตอบได้เลย

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5

ทำไมถึงเป็นtype float เพราะค่า x เป็น int ค่า y เป็น float เมื่อนำมาบวกกันจึงเป็น float

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: 30

ทำไมถึงเป็นtype str เพราะ เลข 30 อยู่ในเครื่องหมาย " " เป็นข้อความ ไม่ใช่ตัวเลข

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: false

ทำไมถึงเป็นtype bool เพราะ is_sunny ถูกกำหนดเป็นค่า True ใน pythom เป็นชนิดข้อมูล boolean

In [ ]:
# Puzzler 4
# What happens here?
num_str = input("Enter a number: ")
result = num_str + 5
print(result)

Enter a number: 55


TypeError: can only concatenate str (not "int") to str

ได้ผลลัพธ์: error

เพราะเป็นการบวกระหว่าง str กับ int จึงเกิด type error